index.js

```diff
@@ -296,12 +296,12 @@ class ExecutorQueue extends Executor {
         if (triggerBuild) {
             config.causeMessage = 'Started by periodic build scheduler';
 
-            return this.postBuildEvent(config)
-                .catch((err) => {
-                    winston.error(`failed to post build event for job ${job.id}: ${err}`);
-
-                    return Promise.resolve();
-                });
+            // Even if post event failed for this event after retry, we should still enqueue the next event
+            try {
+                await this.postBuildEvent(config);
```

Comment:
Should we return after this since `postBuildEvent` will do update job and already go through the enqueue?